components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/NewFormatPedigree.java

```diff
@@ -0,0 +1,120 @@
+    private static final String PEDIGREE_JSON_DATA_KEY = "data";
+
+    /** The name of the property specifying the PhenoTips linked patient id. */
+    private static final String PATIENT_LINK_JSON_KEY = "phenotipsId";
+
+    private static final String PATIENT_LASTNAME_JSON_KEY = "lastName";
+
+    /**
+     * Create a new default pedigree from data (in "simpleJSON" format) and image (a text representing SVG).
+     *
+     * @param data pedigree data
+     * @param image SVG 'image'
+     */
+    public NewFormatPedigree(JSONObject data, String image)
+    {
+        super(data, image);
+    }
+
+    /**
+     * Checks that the provided JSON objects represents a pedigree in the supported format.
+     *
+     * @param data JSON object ot check
+     * @return true if JSON represents a pedigree format supported by this class
+     */
+    public static boolean isSupportedPedigreeFormat(JSONObject data)
+    {
+        return (data.optJSONArray(PEDIGREE_JSON_DATA_KEY) != null);
+    }
+
+    @Override
+    public List<String> extractIds()
+    {
+        // these are only used (for now) when pedigree is saved, and saved pedigrees
+        // do not use new format yet. Gene42 may need this soon though.
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<JSONObject> extractPatientJSONProperties()
+    {
+        // these are only used (for now) when pedigree is saved, and saved pedigrees
+        // do not use new format yet. Gene42 may need this soon though.
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    protected Pair<String, String> getProbandInfo()
+    {
+        JSONArray nodeList = data.optJSONArray(PEDIGREE_JSON_DATA_KEY);
+        if (nodeList != null) {
+            for (Object nodeObj : nodeList) {
+                JSONObject node = (JSONObject) nodeObj;
+                if (node.optBoolean("proband", false)) {
+                    String patientLinkId = node.optString(NewFormatPedigree.PATIENT_LINK_JSON_KEY);
+                    if (!StringUtils.isBlank(patientLinkId)) {
+                        String lastName = node.optString(NewFormatPedigree.PATIENT_LASTNAME_JSON_KEY);
+                        return Pair.of(patientLinkId, lastName);
+                    }
+                    break;
+                }
+            }
+        }
+        return Pair.of(null, null);
+    }
+
+    @Override
+    protected void removeLinkFromPedigreeJSON(String linkedPatientId)
+    {
+        JSONArray nodeList = data.optJSONArray(PEDIGREE_JSON_DATA_KEY);
+        if (nodeList != null) {
+            for (Object nodeObj : nodeList) {
+                JSONObject node = (JSONObject) nodeObj;
+                String nodeLinkId = node.optString(NewFormatPedigree.PATIENT_LINK_JSON_KEY);
+                if (StringUtils.equals(linkedPatientId, nodeLinkId)) {
+                    node.remove(NewFormatPedigree.PATIENT_LINK_JSON_KEY);
```

Comment:
No, the pedigree may not make sense without the node (it may become disconnected, or there may be no partner in a relationship - which is not supported yet. ON top of that it is extremely difficult to do from Java)

..and even if we can solve the all the problems, it still does not make sense e.g. a person (almost) always has two parents, a mother and a father. The fact that father patient profile is gone or unlinked does not remove the fact that there is still a father, which should be recorded in the pedigree